### PR TITLE
BIGC-679 before adding to cart, get the cart first and check whether …

### PR DESCRIPTION
**Pull request by chrisEff**
…products need to be updated or added

### CODE REVIEW

File: extension/lib/shopgate/CartExtensionPipeline.js [hunk 1]
```diff
@@ -27,13 +27,31 @@ class ShopgateCartExtensionPipeline {
...
 
-      return BigCommerceCartRepository.createLineItem(productId, product.quantity, variantId)
-    })
+      let found = null
+      if (bigCommerceCart && bigCommerceCart.lineItems) {
+        found = bigCommerceCart.lineItems._items.find(bcItem => bcItem._productId === productId)
+      }
+      if (found) {
+        itemsToUpdate.push(BigCommerceCartRepository.createLineItemUpdate(found._id, parseInt(product._quantity) + parseInt(found._quantity)))
+      } else {
```
Comment (aaron-martin) [inline extension/lib/shopgate/CartExtensionPipeline.js]: when I tested this I couldn't add another of the same item to the cart. I did some testing and I think there is a problem on line 43. I think product._quantity should be product.quantity. When I console out product at this point { productId: '123', quantity: 1 } is returned.

Comment (chrisEff) [reply extension/lib/shopgate/CartExtensionPipeline.js]: you're right, i fixed that

File: extension/lib/addProductsToCart.js [hunk 2]
```diff
@@ -9,11 +9,21 @@ const { decorateError } = require('./shopgate/logDecorator')
...
+  /*
+   * If we try to add a product that IS already in the cart,
+   * BigC will respond with "500 / Something went wrong",
+   * which can't be distinguished from an actual error in their system.
+   *
+   * So before we ADD, we need to GET the cart first
+   * and then we UPDATE the products which are already present
+   * instead of ADDING them.
+   *
+   * This whole process can take a while and we don't wanna let the user wait for it,
```
Comment (aaron-martin) [inline extension/lib/addProductsToCart.js]: consider changing 'wanna' to 'want to' in the comments.

**Comment by aaron-martin** [comment]
Should there be a change log entry for these changes?

**Pull request merged**